plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2999,9 +3002,18 @@ protected void saveTransOrJob( ISharedObjectsTransformer objectTransformer, Repo
           new RepositoryFile.Builder( file ).title( RepositoryFile.DEFAULT_LOCALE, element.getName() ).createdDate(
               versionDate != null ? versionDate.getTime() : new Date() ).description( RepositoryFile.DEFAULT_LOCALE,
               Const.NVL( element.getDescription(), "" ) ).build();
-      file =
+      try {
+        file =
           pur.updateFile( file, new NodeRepositoryFileData( objectTransformer.elementToDataNode( element ) ),
-              versionComment );
+            versionComment );
+      } catch ( SOAPFaultException e ) {
+        if ( e.getMessage().contains( UnifiedRepositoryUpdateFileException.PREFIX ) ) {
+          throw new KettleException(
+            "File " + file.getName() + " was already updated by another process" );
```

Comment:
Shouldn't this be in the message bundle instead of hard-coded and non-NLS-izable?